pkg/client/client_test.go

```diff
@@ -57,25 +60,47 @@ func TestApplySingle(t *testing.T) {
 		// given
 		namespacedName := types.NamespacedName{Namespace: "toolchain-host-operator", Name: "registration-service"}
 
-		t.Run("when using forceUpdate=true, it should update even when spec is same", func(t *testing.T) {
+		t.Run("when using forceUpdate=true, it should not update when specs are same", func(t *testing.T) {
 			// given
-			cli := NewFakeClient(t)
-			cl := applyCl.NewApplyClient(cli, s)
-			_, err := cl.CreateOrUpdateObject(defaultService.DeepCopyObject(), true, nil)
+			cl, _ := newClient(t, s)
+			obj := defaultService.DeepCopyObject()
+			_, err := cl.CreateOrUpdateObject(obj, true, nil)
 			require.NoError(t, err)
+			originalGeneration := obj.(*corev1.Service).GetGeneration()
```

Comment:
if we use just `defaultService.DeepCopy()` then we don't have to cast it back here and just use `obj.GetGeneration()`

Reply:
no, we can't: `DeepCopy` returns a `runtime.Object` interface which has no suc `GetGeneration()` func :/

Reply:
see my proposal for the new test - I don't have to cast anything there ;-)

Reply:
ahhh, I did not pay enough attention to the methods: you suggested `DeepCopy()` instead of `DeepCopyObject()` (which returns a `runtime.Object` and not a `corev1.Service`)

Reply:
done in 92b60fa